### ISSUE TITLE
Campaign Timeline should start when first email is sent when scheduling campaigns in advance.

### ISSUE THREAD

**Issue by hatsaresilly** [gophish/gophish#999]
What version of Gophish are you using?: 0.5.0

Brief description of the issue: Campaign Timeline should start when first email is scheduled to be sent.

What are you expecting to see happen? : The campaign timeline has a lot of empty space if you schedule campaigns in advance. This makes the chart less readable.

What are you seeing happen?

Please provide as many steps as you can to reproduce the problem:
 * Step 1 - Schedule a campaign to send email in two weeks. 
 * Step 2 - When you get the email click the link and submit the form
 * Step 3 - When looking at the chart, you will see dead space for two weeks, then at the very end you will see where you clicked on the email.
 
 
![image](https://user-images.githubusercontent.com/1242182/37174694-e81539ee-22dc-11e8-8c06-773d8df3fb9d.png)

**Comment by S0larflare** [comment]
I see what you mean, I’ve come across it a couple of times before myself, I just suspect this would take a reasonable amount of work to fix due to how things are currently set up. If I get a few minutes tomorrow I’ll look at how quick this would be to fix

**Comment by jordan-wright** [comment]
I doubt this would take much work at all, it just depends on how we want to handle this.

In terms of the "just works as intended" approach, what do y'all think about adding a "campaign launched" event when the campaign is officially launched? It's debatable about the right place in the code to do this, but I'm sure we'd be able to make it work.

Then, I'd be fine just forcing Javascript to ignore the campaign created event. It seems like something so small that having it be optional is probably a waste of UI space. The "campaign created" event is still there and people can get it from the API if they want.

**Comment by hatsaresilly** [comment]
So, the campaigns page already has the created date, which I think is good. I think this page should also have a Campaign Start Date, and once the campaign is complete, perhaps a completed/ended timestamp as well. Perhaps I should create a new issue for that though. That being said, I like the idea of the graph starting with the launch date, rather than the created date. I think that's a good solution.

**Comment by S0larflare** [comment]
To be fair, this is actually surprisingly easy. 1st example doesn't make a huge impact as the campaign started the following morning after scheduling and ran for a week, but it's enough to show the difference, the 2nd one shows it a bit better

Before
**1**
![image](https://user-images.githubusercontent.com/8939757/37198231-91c5a256-2375-11e8-9519-1698502e6632.png)
**2**
![image](https://user-images.githubusercontent.com/8939757/37198302-c4ff46ea-2375-11e8-8505-b2018ae9c305.png)



After
**1**
![image](https://user-images.githubusercontent.com/8939757/37198193-663e7342-2375-11e8-9df8-932da8477621.png)
**2**
![image](https://user-images.githubusercontent.com/8939757/37198408-1ea96054-2376-11e8-872b-a27e042eb42c.png)




Just by adding one if statement
`if(event.message != "Campaign Created") `

It isn't 100% perfect as there isn't a 'started/launched' event, but I'm not sure it is really needed.

**Comment by S0larflare** [comment]
When you talk about the start date, do you mean on the campaigns page? You can see that already if you hover over the status of the campaign:

![image](https://user-images.githubusercontent.com/8939757/37201721-5e241fda-2380-11e8-9dd4-26b7b1c05b49.png)

**Comment by hatsaresilly** [comment]
Ah, yes. I did not realize the Launch Date was viewable that way. Thanks!

**Comment by jordan-wright** [comment]
All fixed! Great suggestions @hatsaresilly and @S0larflare. I hope this helps!
